torchvision/models/detection/fcos.py

```diff
@@ -0,0 +1,696 @@
+
+        original_image_sizes: List[Tuple[int, int]] = []
+        for img in images:
+            val = img.shape[-2:]
+            assert len(val) == 2
+            original_image_sizes.append((val[0], val[1]))
+
+        # transform the input
+        images, targets = self.transform(images, targets)
+
+        # Check for degenerate boxes
+        if targets is not None:
+            for target_idx, target in enumerate(targets):
+                boxes = target["boxes"]
+                degenerate_boxes = boxes[:, 2:] <= boxes[:, :2]
+                if degenerate_boxes.any():
+                    # print the first degenerate box
+                    bb_idx = torch.where(degenerate_boxes.any(dim=1))[0][0]
+                    degen_bb: List[float] = boxes[bb_idx].tolist()
+                    raise ValueError(
+                        "All bounding boxes should have positive height and width."
+                        f" Found invalid box {degen_bb} for target at index {target_idx}."
+                    )
+
+        # get the features from the backbone
+        features = self.backbone(images.tensors)
+        if isinstance(features, torch.Tensor):
+            features = OrderedDict([("0", features)])
+
+        features = list(features.values())
+
+        # compute the fcos heads outputs using the features
+        head_outputs = self.head(features)
+
+        # create the set of anchors
+        anchors = self.anchor_generator(images, features)
+        # recover level sizes
+        num_anchors_per_level = [x.size(2) * x.size(3) for x in features]
+
+        losses = {}
+        detections: List[Dict[str, Tensor]] = []
+        if self.training:
+            assert targets is not None
+
+            # compute the losses
+            losses = self.compute_loss(targets, head_outputs, anchors, num_anchors_per_level)
+        else:
+            # split outputs per level
+            split_head_outputs: Dict[str, List[Tensor]] = {}
+            for k in head_outputs:
+                split_head_outputs[k] = list(head_outputs[k].split(num_anchors_per_level, dim=1))
+            split_anchors = [list(a.split(num_anchors_per_level)) for a in anchors]
+
+            # compute the detections
+            detections = self.postprocess_detections(split_head_outputs, split_anchors, images.image_sizes)
+            detections = self.transform.postprocess(detections, images.image_sizes, original_image_sizes)
+
+        if torch.jit.is_scripting():
+            if not self._has_warned:
+                warnings.warn("FCOS always returns a (Losses, Detections) tuple in scripting")
+                self._has_warned = True
+            return losses, detections
+        return self.eager_outputs(losses, detections)
+
+
+model_urls = {
+    "fcos_resnet50_fpn_coco": "https://github.com/o295/checkpoints/releases/download/coco/fcos_resnet50_fpn_coco-7c2e6686.pth",
+}
+
+
+def fcos_resnet50_fpn(
```

Comment:
Blocking: We need the equivalent builder on the prototype area for the new API.

@xiaohu2015 @zhiqwang I'm happy to catch up with you and help you do it. I'm also happy to push the changes for you if you are busy. But I would really love it if you have a look and try to implement it yourself based on the existing examples ([old](https://github.com/pytorch/vision/blob/main/torchvision/models/detection/faster_rcnn.py) vs [new](https://github.com/pytorch/vision/blob/main/torchvision/prototype/models/detection/faster_rcnn.py)). You would be the first people to implement the new API outside of the maintaining team so your feedback would be VERY interesting. Let me know. :)

Reply:
Hi @datumbox , I'll try to implement the prototype builder for loading fcos !

Reply:
Awesome mate, thanks a lot! Looking forward to your input.

Reply:
Hi @datumbox , We've added the prototype builder for `fcos`. Because the interface between `fcos` and `retinanet` is very similar, this builder can be implemented very quickly.

I love this prototype API, It's feature-rich and easy to use with the help of well-designed tools!

Reply:
Glad you found it useful mate. And thanks to both you and @xiaohu2015 for test driving our new process, APIs etc. It really helps us streamline them. Let's discuss after the PR concludes to get your input on how to improve the experience.